src/Magnum/Octree/Octree.hpp

```diff
@@ -0,0 +1,293 @@
+                if(octContainingOldBox != -1) {
+                    _children[octContainingOldBox]->insert(oldPoint, oldData);
+                    _entries.erase(_entries.begin());
+                    _data.erase(_data.begin());
+                }
+
+                if(octContainingBox != -1) {
+                    _children[octContainingBox]->insert(point, data);
+                } else {
+                    _entries.push_back(point);
+                    _data.push_back(data);
+                }
+            } else {
+                _entries.push_back(point);
+                _data.push_back(data);
+            }
+        }
+    } else {
+        /* Node is an inner node. Insert recursively into the appropriate child octant */
+        const Int octContainingBox = getOctantContainingBox(point);
+
+        if(octContainingBox != -1) {
+            _children[octContainingBox]->insert(point, data);
+        } else {
+            _entries.push_back(point);
+            _data.push_back(data);
+        }
+    }
+
+    return *this;
+}
+
+/* Determine which octant of the octree would contain 'point' */
+template <typename T>
+int Octree<T>::getOctantContainingBox(const Range3D& point) const {
+    Int oct = 0;
+
+    /* If one of max.x/y/z is greater than the center.x/y/z, but min.x/y/z is less than center.x/y/z,
+     * the AABB doesn't fit into any child node of this node, it therefore has to be stored
+     * into this node (-1 is returned).
+     */
+    if(point.min().x() >= _center.x()) {
+        oct |= 1;
+    } else if(point.max().x() > _center.x()) {
+        return -1;
+    }
+
+    if(point.min().y() >= _center.y()) {
+        oct |= 2;
+    } else if(point.max().y() > _center.y()) {
+        return -1;
+    }
+
+    if(point.min().z() >= _center.z()) {
+        oct |= 4;
+    } else if(point.max().z() > _center.z()) {
+        return -1;
+    }
+
+    return oct;
+}
+
+/* Return all points in the octree contained in the bounds spanned by min, max */
+template <typename T>
+Octree<T>& Octree<T>::points(std::vector<T>& resultData, const Vector3& min, const Vector3& max) {
+    for(UnsignedInt i = 0; i < _data.size(); ++i) {
+        if((Math::min(min, _entries.at(i).min()) == min) && (Math::max(max, _entries.at(i).max()) == max)) {
+            resultData.push_back(_data.at(i));
+        }
+    }
+
+    if(!isLeafNode()) {
+        for(UnsignedInt i = 0; i < 8; ++i) {
+            /* Compute the min/max corners of this child octant */
+            Octree<T>& child = *_children[i];
+            const Vector3 cmin = child._center - Vector3{child._radius};
+            const Vector3 cmax = child._center + Vector3{child._radius};
+
+            /* If query rectangle doesn't intersect this child octants bounding box, continue. */
+            if(Math::max(max, cmin) != max) {
+                continue;
+            }
+            if(Math::min(min, cmax) != min) {
+                continue;
+            }
```

Comment:
Couldn't `min` and `max` be represented as `Range3D` and then using some potential `Range3D::contains()` / `Range3D::overlaps()` functionality to get what you need?